src/analysis/rust_type.rs

```diff
@@ -230,12 +235,19 @@ pub fn rust_type_full(
                 if p.closure.is_some() {
                     continue
                 }
-                match rust_type(env, p.typ) {
+                match rust_type_nullable(env, p.typ, p.nullable) {
                     Ok(x) => {
                         let is_fundamental = p.typ.is_fundamental_type(env);
+                        let y = rust_type(env, p.typ).unwrap_or_else(|_| String::new());
                         s.push(format!("{}{}",
                                        if is_fundamental { "" } else { "&" },
-                                       if x != "GString" { x } else { "&str".to_owned() }));
+                                       if y != "GString" { x } else {
```

Comment:
Maybe just `!ends_with("GString")` ?

Reply:
I don't think this is a good idea: if we ever have a type that for any reason ends with "GString" (because why not?), well, debugging will be complicated. :)